util/jsonresponse.py

```diff
@@ -53,6 +54,134 @@ def json(self):
 
         return json.dumps(out, indent=2, cls=NumpyJSONEncoder)
 
+    @log_timing(log)
+    def write_json(self, path):
+        file_paths = []
+        base_path = os.path.join(app.config['LOCAL_ASYNC_DIR'], path)
+
+        if not os.path.isdir(base_path):
+            try:
+                os.makedirs(base_path)
+            except OSError:
+                if not os.path.isdir(base_path):
+                    raise WriteErrorException('Unable to create local output directory: %s' % path)
+
+        # annotation data will be written to a JSON file
+        if self.stream_request.include_annotations:
+            time_range_string = str(self.stream_request.time_range).replace(" ", "")
+            anno_fname = 'annotations_%s.json' % (time_range_string)
+            anno_json = os.path.join(base_path, anno_fname)
+            file_paths.append(anno_json)
+            self.stream_request.annotation_store.dump_json(anno_json)
+
+        stream_key = self.stream_request.stream_key
+        stream_dataset = self.stream_request.datasets[stream_key]
+        parameters = self.stream_request.requested_parameters
+        external_includes = self.stream_request.external_includes
+        for deployment, ds in stream_dataset.datasets.iteritems():
+            refdes = stream_key.as_dashed_refdes()
+            times = ds.time.values
+            start = ntp_to_short_iso_datestring(times[0])
+            end = ntp_to_short_iso_datestring(times[-1])
+
+            # provenance types will be written to JSON files
+            if self.stream_request.include_provenance:
+                prov_fname = 'deployment%04d_%s_provenance_%s-%s.json' % (deployment,
+                                                                          stream_key.as_dashed_refdes(), start, end)
+                prov_json = os.path.join(base_path, prov_fname)
+                file_paths.append(prov_json)
+                stream_dataset.provenance_metadata.dump_json(prov_json)
+
+            filename = 'deployment%04d_%s_%s-%s.json' % (deployment, refdes, start, end)
```

Comment:
for provenance we used `stream_key.as_dashed_refdes()`, here we use `refdes`. Should it be consistent?

Reply:
Changing to use stream_key.as_dashed_refdes() in both places. Making change in jsonresponse.py and in csvresponse.py